File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty()) return 0;
        int n = nums[0];
        int idx = 1;
        int count = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] > n) {
                count++;
                nums[idx] = nums[i];
                n = nums[i];
                idx++;
            }
        }
        return count;
    }
};

File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if (len(nums) <= 1):
            return len(nums)
        prev = nums[0]
        idx = 1
        count = 1
        for i in range(1,len(nums)):
            if (nums[i] <= prev):
                continue
            prev = nums[i]
            tmp = nums[idx]
            nums[idx] = nums[i]
            nums[i] = tmp
            idx += 1
            count += 1
        return count
            